ui/app/templates/components/shamir-modal-flow.hbs

```diff
@@ -0,0 +1,194 @@
+        {{#if otp}}
+          <h4 class="title is-7 has-bottom-padding-m is-fullwidth">
+            One time password (OTP)
+          </h4>
+          <div class="message is-list has-copy-button" tabindex="-1">
+            <HoverCopyButton @copyValue={{otp}} />
+            <code class="is-word-break">{{otp}}</code>
+          </div>
+        {{/if}}
+          <h4 class="title is-7 has-bottom-padding-m is-fullwidth">
+            DR operation token command
+          </h4>
+          <div class="message is-list has-copy-button" tabindex="-1">
+            {{#let (if otp
+              (concat 'vault operator generate-root -otp="' otp '" -decode="' encoded_token '"') (concat 'vault operator generate-root -otp="<enter your otp here>" -decode="' encoded_token '"') ) as |cmd|}}
+              <HoverCopyButton @copyValue={{cmd}} />
+              <code class="is-word-break">{{cmd}}</code>
+            {{/let}}
+          </div>
+      </div>
+      <div>
+        <button type="button" class="button is-primary" {{action 'onCancelClose'}}>
+          Clear &amp; Close
+        </button>
+      </div>
+    {{else if (and generateAction (not started))}}
+    <form {{action 'startGenerate' (hash pgp_key=pgp_key) on="submit"}} id="shamir">
+      {{message-error errors=errors}}
+      {{#if (eq generateStep 'chooseMethod')}}
+        <div class="has-bottom-margin-m" data-test-shamir-modal-body>
+          {{yield}}
+        </div>
+        <div class="field is-grouped">
+          <div class="control is-flex-row">
+            <button type="button" class="link" {{action (mut generateWithPGP) true}}>
+              Provide PGP Key
+            </button>
+          </div>
+          <div class="control">
+            <span class="button auto-width is-white is-static">
+              or
+            </span>
+          </div>
+          <div class="control">
+            <button type="submit" class="button is-primary" data-test-generate-token-cta>
+              Generate operation token
+            </button>
+          </div>
+        </div>
+      {{/if}}
+      {{#if (eq generateStep 'providePGPKey')}}
+        <div class="has-bottom-margin-m">
+          <p>
+            Choose a PGP Key from your computer or paste the contents of one in the form below.
+            This key will be used to Encrypt the generated operation token.
+          </p>
+          {{pgp-file index='' key=pgpKeyFile onChange=(action 'setKey')}}
+        </div>
+        <div class="field is-grouped">
+          <div class="control">
+            <button type="button" class="button" {{action "reset"}}>
+              Back
+            </button>
+          </div>
+          <div class="control">
+            <button type="button" disabled={{not pgp_key}} class="button is-primary" {{action "savePGPKey"}}>
+              Use PGP Key
+            </button>
+          </div>
+        </div>
+      {{/if}}
+      {{#if (eq generateStep 'beginGenerationWithPGP')}}
+        <div>
```

Comment:
i remember having this thought last time i was working in this area of the codebase, but it would be so nice to refactor this template. maybe pulling out each `generateStep` into it's own component? totally out of scope of this PR, but just a nice thought to daydream about.